packages/babel-preset/src/transformPlugin.ts

```diff
@@ -96,9 +96,17 @@ function isRequireDeclarator(
  */
 function dedupeCSSRules(cssRules: CSSRulesByBucket): CSSRulesByBucket {
```

Comment:
We can do it better. Let's move it to `packages/babel-preset/src/utils/dedupeRules.ts`
```ts
  export function dedupeCSSRules(cssRules: CSSRulesByBucket): CSSRulesByBucket {
    return Object.fromEntries(
      Object.entries(cssRules).map(([bucketName, cssRules]) => {
        if (Array.isArray(cssRules)) {
          return [bucketName, [...new Set(cssRules)]];
        }

        return [bucketName, dedupeCSSRules(cssRules)];
      }),
    );
  }
```

```ts
import { dedupeCSSRules } from './dedupeRules';

describe('dedupeCSSRules', () => {
  it('de-duplicates rules', () => {
    expect(dedupeCSSRules({ d: ['foo', 'bar'], k: ['foo', 'baz'] })).toEqual({ d: ['foo', 'bar'], k: ['foo', 'baz'] });
  });

  it('de-duplicates rules for media queries', () => {
    expect(dedupeCSSRules({ m: { foo: ['foo', 'bar'], baz: ['baz'] } })).toEqual({
      m: { foo: ['foo', 'bar'], baz: ['baz'] },
    });
  });
});

```

Reply:
I would prefer to keep the check on the bucket name to be very clear that this only happens for media queries

Reply:
That could also work, not sure if TS will like it.